src/components/organisms/SettingsDropDown.tsx

```diff
@@ -0,0 +1,242 @@
+  width: "100%",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "flex-start",
+});
+
+const Label = styled("label", {
+  fontSize: 13,
+  lineHeight: 1,
+  marginBottom: 10,
+  color: indigo.indigo12,
+  display: "block",
+});
+
+const Input = styled("input", {
+  all: "unset",
+  flex: "1 0 auto",
+  borderRadius: 4,
+  padding: "0 10px",
+  fontSize: 15,
+  lineHeight: 1,
+  color: indigo.indigo11,
+  boxShadow: `0 0 0 1px ${indigo.indigo7}`,
+  height: 35,
+  "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo8}` },
+  "&::placeholder": { color: mauve.mauve7 },
+});
+
+const FieldCheck = styled("fieldset", {
+  all: "unset",
+  width: "100%",
+  display: "flex",
+  alignItems: "center",
+});
+
+const CheckLink = styled("a", {
+  color: indigo.indigo11,
+  textDecoration: "underline",
+  "&:visited": { color: indigo.indigo11 },
+});
+
+type Props = {
+  setSettingsView: React.Dispatch<React.SetStateAction<string>>;
+  settingsView: React.Dispatch<React.SetStateAction<string>>;
+  setSelectedDeviceName: React.Dispatch<React.SetStateAction<string>>;
+  selectedDeviceName: React.Dispatch<React.SetStateAction<string>>;
+  selectedDeviceIndex: React.Dispatch<React.SetStateAction<number>>;
+  setSelectedDeviceIndex: React.Dispatch<React.SetStateAction<number>>;
+  setDeviceData: React.Dispatch<React.SetStateAction<[]>>;
+  deviceData: React.Dispatch<React.SetStateAction<[]>>;
+  isLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export default function DropDownSettings({
+  setSettingsView,
+  settingsView,
+  setSelectedDeviceName,
+  selectedDeviceName,
+  selectedDeviceIndex,
+  setSelectedDeviceIndex,
+  deviceData,
+  setDeviceData,
+  isLoading,
+  setIsLoading,
+}: Props) {
+  const getDeviceName = () => {
+    let dropDownText = document.getElementById("drop_menu_settings");
+    let selectedText = dropDownText.options[dropDownText.selectedIndex].text;
+    return setSelectedDeviceName(selectedText);
+  };
+
+  // useEffect(() => {
+  //   const index: number = deviceData.findIndex((e: {}) => e.device_name === selectedDeviceName);
+  //   console.log(index);
+  //   setSelectedDeviceIndex(index);
+  // }, [deviceData])
+
+  const settingsViewChange = () => {
+    getDeviceName();
+    setSettingsView("read settings");
+    // console.log("DROPDOWN", deviceData);
+    console.log("DROPDOWN", selectedDeviceIndex);
+  };
+
+  if (isLoading === true) {
+    return <div className="loading">Loading...</div>;
+  } else {
+    return (
+      <>
+        <div className={styles.top}>
+          <div className={styles.top__inner}></div>
+          <div className="select_device">
+            <h3>Please select your device:</h3>
+            <select className="drop_menu" id="drop_menu_settings">
+              {deviceData.map((obj: any) => {
+                return (
+                  <option value={obj.device_name}>{obj.device_name}</option>
```

Comment:
When using loop, you need to add `key`.
ex. `key={i}`

Reply:
Done